public/friendlycode/js/fc/ui/bramble-proxy.js

```diff
@@ -70,6 +71,55 @@ define(["backbone-events"], function(BackboneEvents) {
     }
   }
 
+  
+  BrambleProxy.prototype.undo = function () {
+    this.onButton("_undo");
+  };
+  BrambleProxy.prototype.redo = function () {
+    this.onButton("_redo");
+  };
+  /* This function handles all the button presses thimble has,
+   * It takes in 2 parameters, a string representing the command, and a possible option
+   * It packages a JSON object and sends it over post to thimble proxy inside brackets
+   * jType refers to what type of command should be fired inside brackets, these types being:
+   *    mCommand: Menu Command, refers to any command outlined in the menu
+   *    vCommand: View Command, refers to commands inside the ViewHandler
+   * jParams is used in conjunction with vCommand to send extra paramters needed for vCommand
+   */
+  BrambleProxy.prototype.onButton = function(command, option) {
```

Comment:
This protocol isn't very semantic. Shortening names isn't very helpful, especially with modern text editors that can autocomplete common phrases.

Go for identifiers (especially in the protocol itself) that give some obvious clues about the purpose of the thing being identified.